_ingest-pipelines/processors/foreach.md

```diff
@@ -0,0 +1,86 @@
+---
+layout: default
+title: Foreach
+parent: Ingest processors
+nav_order: 110
+---
+
+# Foreach processor
+
+The `foreach` processor is used to <explain what is used to do>.
+
+The following is the syntax for the `foreach` processor:
+
+```json
+<insert syntax example>
+```
+{% include copy-curl.html %}
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `foreach` processor.
+
+Parameter | Required/Optional | Description |
+|-----------|-----------|-----------|
+<insert the parameters>
+
+## Using the processor
+
+Follow these steps to use the processor in a pipeline.
+
+### Step 1: Create a pipeline
+
+The following query creates a pipeline, named <name>, that uses the `foreach` processor to <do what?>: 
+
+```json
+<insert pipeline code example>
+```
```

Comment:
@gaobinlong When I test this above example in Dev Tools, I'm getting the following error. Will you provide a valid pipeline example? 
```json
{
  "error": {
    "root_cause": [
      {
        "type": "parse_exception",
        "reason": "[processors] required property is missing",
        "property_name": "processors"
      }
    ],
    "type": "parse_exception",
    "reason": "[processors] required property is missing",
    "property_name": "processors"
  },
  "status": 400
}
```